src/screens/Auth/EmailVerify/EmailVerify.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+
+import CheckCircleOutlineOutlinedIcon from "@material-ui/icons/CheckCircleOutlineOutlined";
+
+import Button from "../../../Components/Button/Button.jsx";
+
+import useHumanizer from "../../../hooks/useHumanizer.js";
+import useSnack from "../../../hooks/useSnack.js";
+import { getProfile, requestEmailVerification } from "../../../utils/api.js";
+
+import "./EmailVerify.scss";
+
+const EmailVerify = ({
+  onNext = () => {},
+  serverInfo,
+  user,
+  setUser = () => null,
+}) => {
+  const { t } = useTranslation();
+  const { showSnack } = useSnack();
+
+  const dontReceived = t("screens.emailVerify.dontReceivedEmail");
+  const [dontReceivedBefore, dontReceivedLink, ...dontReceivedAfter] = useMemo(
+    () => dontReceived.split(/[[\]]/),
+    [dontReceived]
+  );
+  const { durationHumanizer } = useHumanizer(t);
+  const secondsLeft = useMemo(
+    () =>
+      (new Date(user?.user?.createdAt).getTime() +
+        serverInfo?.email_confirm?.time -
+        Date.now()) /
+      1000,
+    [serverInfo?.email_confirm?.time, user?.user?.createdAt]
+  );
+
+  // divide by an hour and round to exclude seconds and minutes
+  const time = durationHumanizer({
+    duration: ~~(secondsLeft / 3600) * 3600,
+  });
+
+  const resend = useCallback(() => {
+    requestEmailVerification(
+      { email: user.user.email },
+      {
+        headers: {
+          Authorization: `Bearer ${user.token}`,
+        },
+      }
+    )
+      .then(() => {
+        showSnack("success", t("screens.emailVerify.resendEmailSuccess"));
+      })
+      .catch(() => {
+        showSnack("error", t("screens.emailVerify.resendEmailError"));
+      });
+  }, [showSnack, t, user?.token, user?.user?.email]);
+
+  const checkVerificationStatus = useCallback(async () => {
+    try {
+      const profileRes = await getProfile({
+        headers: {
+          Authorization: `Bearer ${user?.token}`,

```

Comment:
Hasn't the authorization token not already been set in the api.js?
Or is this check running externally?

Reply:
I see. Then this should be fine too